frontend/js/src/forms/ProfileEdit.tsx

```diff
@@ -0,0 +1,115 @@
+import { Formik } from "formik";
+import React, { JSX } from "react";
+import { createRoot } from "react-dom/client";
+import * as Yup from "yup";
+import { getPageProps } from "../utils";
+import { Dataset, DatasetsInput, TextInput } from "./utils";
+
+type ProfileEditProps = {
+  datasets: Dataset[];
+  is_commercial: boolean;
+  csrf_token: string;
+  initial_form_data: any;
+  initial_errors: any;
+};
+
+function ProfileEdit({
+  datasets,
+  is_commercial,
+  csrf_token,
+  initial_form_data,
+  initial_errors,
+}: ProfileEditProps): JSX.Element {
+  return (
+    <>
+      <h1 className="page-title">Your Profile</h1>
+      <h2>Edit contact information</h2>
+
+      <Formik
+        initialValues={{
+          datasets:
+            initial_form_data.datasets?.map((x: number) => x.toString()) ?? [],
+          contact_name: initial_form_data.contact_name ?? "",
+          contact_email: initial_form_data.contact_email ?? "",
+          csrf_token,
+        }}
+        initialErrors={initial_errors}
+        initialTouched={initial_errors}
```

Comment:
Hi! Yes, this is intentional because of the way our routing works. We are not a SPA so the page is reloaded and the errors are returned in the props of the new HTML so the errors returned by backend will by populated in initial errors and those fields need to marked as touched to show the errors correctly.